### ISSUE TITLE
How to resume_from_checkpoint for Seq2SeqTrainer of EncoderDecoderLM

### ISSUE THREAD

**Issue by DapangLiu** [huggingface/transformers#11113]
Hi transformers developers:

First thanks to your great work!

I noticed there was a resume from checkpoint method for the Seq2SeqTrainer when the LM was BART-like out-of-the-box Seq2Seq LMs, like this:

https://github.com/huggingface/transformers/blob/fd338abdeba25cb40b27650ba2203ac6789d2776/examples/seq2seq/run_summarization.py#L529

I wonder how can I use it for a [EncoderDecoderLM](https://huggingface.co/transformers/model_doc/encoderdecoder.html), where the whole model has two parts (encoder and decoder). Should I load the checkpoint for encoder and decoder separately, or is there a nice way we can do that in one line like BART?

Thanks in advance!

**Comment by patil-suraj** [comment]
Hi @DapangLiu 

The `resume_from_checkpoint` should work for any `PreTrainedModel` class. Even though `EncoderDecoder` model is initialized using two sepearte models when save it using `save_pretarined` it's saved as a single model and can be loaded using `from_pretrained`.

Make sure that your checkpoint directory contains the saved model file, the saved optimizer and scheduler (if you also want to resume that) and it should work.

**Comment by DapangLiu** [comment]
Solved my problem. Thanks @patil-suraj !

**Comment by helloworld123-lab** [comment]
Hello everyone,
I have a similar problem with save&load models. In my model's training_arguments, I assign save_steps and it saves the last checkpoint including optimizer.pt, scheduler.pt, pytorch_model.bin etc.

I load back my model as:

```
from transformers import BertTokenizer, EncoderDecoderModel, EncoderDecoderConfig

tokenizer = BertTokenizer.from_pretrained("../input/checkpoint-21900/")
encoder_decoder_config = EncoderDecoderConfig.from_pretrained('../input/checkpoint-21900/')
bert2bert = EncoderDecoderModel.from_pretrained("../input/checkpoint-21900/", config=encoder_decoder_config)
bert2bert.to("cuda")
```

Then, I start the training process of the loaded model with the same training arguments. However, in the first model the losses for the final steps are: 

```
21600 | 0.424700
21700 | 0.393000
21800 | 0.446200
21900 | 0.391600
```

In the second model, for the new training process the training loss values are increasing as:
```

Step | Training Loss
-- | --
100 | 0.500100
200 | 0.589900
300 | 0.564400
400 | 0.557400
500 | 0.564600
600 | 0.590800
700 | 0.661900
```
 Am I doing something wrong in the loading procedures of the first model? Because it is expected to decrease the loss with new epochs?
